src/screens/mainScreen.h

```diff
@@ -22,12 +23,15 @@ class ScreenMainScreen : public GuiCanvas, public Updatable
     GuiRadarView* long_range_radar;
     bool first_person;
     GuiCommsOverlay* onscreen_comms;
+    Music* music;
```

Comment:
instead of using raw pointer `Music*` use a `std::unique_ptr<Music>` as right now, you leak memory when the main screen is destroyed, as the Music object is never destroyed. Instead of manually destroying it, you can just use `std::unique_ptr` to delete it for you.

You might notice I use raw pointers for the GUI elements, because those are owned by their parent GUI element, so they will get destroyed when the GUI is destroyed.

Reply:
coincidentally learned about shared_ptr and unique_ptr last night so well timed, thanks @daid